raiden/network/proxies/token_network.py

```diff
@@ -0,0 +1,773 @@
+
+        if not channel_identifier:
+            log.error(
+                'channel_identifier failed',
+                peer1=pex(self.node_address),
+                peer2=pex(other_peer)
+            )
+            raise RuntimeError('channel_identifier failed')
+
+        log.info(
+            'new_netting_channel called',
+            peer1=pex(self.node_address),
+            peer2=pex(other_peer),
+            channel_identifier=channel_identifier,
+        )
+
+        return channel_identifier
+
+    def _new_netting_channel(self, other_peer, settle_timeout):
+        if self.channel_exists(other_peer):
+            raise DuplicatedChannelError('Channel with given partner address already exists')
+
+        transaction_hash = estimate_and_transact(
+            self.proxy,
+            'openChannel',
+            self.node_address,
+            other_peer,
+            settle_timeout,
+        )
+
+        if not transaction_hash:
+            raise RuntimeError('open channel transaction failed')
+
+        self.client.poll(unhexlify(transaction_hash), timeout=self.poll_timeout)
+
+        if check_transaction_threw(self.client, transaction_hash):
+            raise DuplicatedChannelError('Duplicated channel')
+
+        return transaction_hash
+
+    def channel_exists(self, partner: typing.Address) -> bool:
+        channel_data = self.detail_channel(partner)
+
+        if not isinstance(channel_data['state'], typing.T_ChannelState):
+            raise ValueError('channel state must be of type ChannelState')
+
+        log.debug('existing_channel {}'.format(channel_data))
+
+        return channel_data['state'] > CHANNEL_STATE_NONEXISTENT
+
+    def _detail_participant(self, participant: typing.Address, partner: typing.Address):
+        """ Returns a dictionary with the channel participant information. """
+        data = self._call_and_check_result('getChannelParticipantInfo', participant, partner)
+        return {
+            'deposit': data[0],
+            'withdrawn': data[1],
+            'is_closer': data[2],
+            'balance_hash': data[3],
+            'nonce': data[4],
+        }
+
+    def detail_channel(self, partner: typing.Address):
+        """ Returns a dictionary with the channel specific information. """
+        channel_data = self._call_and_check_result('getChannelInfo', self.node_address, partner)
+        return {
+            'channel_identifier': channel_data[0],
+            'settle_block_number': channel_data[1],
+            'state': channel_data[2],
+        }
+
+    def detail_participants(self, partner: typing.Address):
+        """ Returns a dictionary with the participants' channel information. """
+        our_data = self._detail_participant(self.node_address, partner)
+        partner_data = self._detail_participant(partner, self.node_address)
+        return {
+            'our_address': self.node_address,
+            'our_balance': our_data['deposit'],
```

Comment:
Since this is the smart contract shouldn't this also be `our_deposit` and not `balance`? Since `balance` is the actual balance when taking into account the `transferred_amount` while `deposit` is what the smart contract knows is our deposit to the contract.

Reply:
Generally, I am unsure on how to proceed with interface names. Renaming things that used in the rest of the codebase will make it harder to do the final integration. This is why I tried to keep  `our_balance` and most of the function names as they were and only use updated names within the function logic.
So:
1) what should be my general approach? 
2) keep `our_balance` or use `our_deposit`?

Reply:
I would say `our_deposit` and as discussed in RC change here now -- document the change and keep it in mind for the further integration